### PR TITLE
net35 task host properly handle isolated AppDomain tasks

### PR DESCRIPTION
**Pull request by rokonec**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1729428

### Context
Change https://github.com/dotnet/msbuild/pull/7387/files#diff-2295212a8e8b80807c0594f9e600d287e5a813c10326d71b4ac5ce907f0ecc42R45 have caused regression of not running specific net35 task in isolated AppDomain.

In our particular case of customer feedback issue, `UnregisterAssembly` task have kept some assemblies locked, because it need to load assemblies in order to unregister them, while prior this changes assembly files were released when `AppDomain` which loaded them have been deleted.

### Changes Made
Fallback to old behavior handling detecting attributes from full types as oppose to reflection only types.
Changes was based on changes of related PR ^.

### Testing
I have created repro. After fixes it no longer repros.

### Notes
Consequences of this error might be serious. Everyone with VS 17.3+ compiling and registering COM objects in net35 will experience it while rebuilding. We do not have multiple reports of this issue, AFAIK, but people might hesitate to report it, as obvious workarounds  exists (rebuild again, or use older VS 2019-).

@rainersigwald  we still might consider to backport it and/or servicing it in VS 17.5

### CODE REVIEW

**Review by Forgind — APPROVED** [review]
I don't think this will break anything, and it sounds like it works, but I'm not sure this is the right fix. If the bug only reproduces every other build, that clearly means the task host is dying every other build. Why is that true? Task hosts must load assemblies, hence locking the assembly, to do their work. Then they should die. They should not be retaining locks, no matter how the task was loaded, because they should not exist after they're finished. Am I wrong about that?

**Comment by rokonec** [comment]
@Forgind I am positive it is the right fix. 
Here is how that `every other build failed` was happening:
  1. rebuild, it builds an assembly which is registered as COM by RegisterAssembly, during that registration it creates a cache file in obj for example `obj\some.csproj.UnmanagedRegistration.cache` for purpose of being able to clean it properly if needed
  2. next rebuild, during clean it finds `obj\some.csproj.UnmanagedRegistration.cache` which means it has to unregister assembly listed in that cache. This has to be done by `msbuildtaskhost` for it is net35 COM object. Task succeed, assembly is unregistered and cache file is deleted. However, when same build need to access the assembly file for any reason, copy mostly, the file is still locked and build failed.
  3. next build, during clean it does NOT find  `obj\some.csproj.UnmanagedRegistration.cache`  as it has been deleted by previous rebuild and succeed
  4. goto 2.

In other words, that lock is not hold be previous build. The latter rebuild locks it itself.

Task host is actually dying every single build, but after whole build is finished (failed or not).

**Review by ladipro — APPROVED** [review]
nit: Can you please also fix the comment in the `#if !NET35` block at line 45? It does not seem to be true.

  `// Properties set in this block aren't used by TaskHosts..`
